lightningd/peer_htlcs.c

```diff
@@ -491,23 +493,22 @@ static void forward_htlc(struct htlc_in *hin,
 	}
 
 	/* Fee base and ppm can also be set per channel */
-	if (next->feerate_base > 0) {
+	if (next->feerate_base > 0)
 		fee_base = next->feerate_base;
-	}
-	// TODO: next->feerate_ppm
+	if (next->feerate_ppm > 0)
+		fee_ppm = next->feerate_ppm;
 
 	/* BOLT #7:
 	 *
 	 * The origin node:
 	 *   - SHOULD accept HTLCs that pay a fee equal to or greater than:
 	 *     - fee_base_msat + ( amount_to_forward * fee_proportional_millionths / 1000000 )
 	 */
-	if (mul_overflows_u64(amt_to_forward, ld->config.fee_per_satoshi)) {
+	if (mul_overflows_u64(amt_to_forward, fee_ppm)) { // TODO: why is /= 1000000 missing here?
```

Comment:
Most likely because the `/ 10**6` is applied after the multiplication would already have overflown.